topo/tarjan.go

```diff
@@ -31,24 +31,46 @@ func (e Unorderable) Error() string {
 	return fmt.Sprintf("topo: no topological ordering: cyclic components: %v", [][]graph.Node(e))
 }
 
+func lexical(nodes []graph.Node) { sort.Sort(ordered.ByID(nodes)) }
+
 // Sort performs a topological sort of the directed graph g returning the 'from' to 'to'
 // sort order. If a topological ordering is not possible, an Unorderable error is returned
 // listing cyclic components in g with each cyclic component's members sorted by ID. When
 // an Unorderable error is returned, each cyclic component's topological position within
 // the sorted nodes is marked with a nil graph.Node.
 func Sort(g graph.Directed) (sorted []graph.Node, err error) {
 	sccs := TarjanSCC(g)
-	sorted = make([]graph.Node, 0, len(sccs))
+	return finalizeSort(sccs, lexical)
+}
+
+// SortStabilized performs a topological sort of the directed graph g returning the 'from'
```

Comment:
I would add something along the lines of "order, when called, should re-order the nodes in-place".